services/controlplane/client_test.go

```diff
@@ -342,3 +343,61 @@ func TestRetriesTimeout(t *testing.T) {
 		})
 	}
 }
+
+func TestGetDestinationSSHKeyPair(t *testing.T) {
```

Comment:
@saurav-malani I converted some test checks to make use of `require` for a more succint test.